### PR TITLE
Use better asserts. Added notes to style guide.

### PR DESCRIPTION
**Pull request by atombrella**
@bmw Please check this. I think it improves readability quite substantially for some of the statements.

I allowed myself to include a small note about `editorconfig` as well.

## Pull Request Checklist

- [x] If the change being made is to a [distributed component](https://certbot.eff.org/docs/contributing.html#code-components-and-layout), edit the `master` section of `certbot/CHANGELOG.md` to include a description of the change being made.
- [x] Include your name in `AUTHORS.md` if you like.

### CODE REVIEW

**Comment by adferrand** [comment]
Since this PR is a response to the discussion we had on the recent changes for ECDSA certificats, maybe you meant to hightlight me instead of bmw?

**Comment by adferrand** [comment]
Nevermind, I just saw it also concerns #8389

**Comment by atombrella** [comment]
@adferrand Yes. I made a mistake with the tagging. However, maybe it's better to deal with these asserts in smaller batches.

File: certbot-apache/tests/obj_test.py [hunk 1]
```diff
@@ -27,14 +27,14 @@ def test_repr(self):
             "certbot_apache._internal.obj.Addr(('127.0.0.1', '443'))")
 
     def test_eq(self):
-        self.assertTrue(self.vhost1b == self.vhost1)
-        self.assertFalse(self.vhost1 == self.vhost2)
+        self.assertEqual(self.vhost1b, self.vhost1)
+        self.assertEqual(self.vhost1, self.vhost2)
```
Comment (adferrand) [inline certbot-apache/tests/obj_test.py:31]: This should be `assertNotEqual`. And I am surprised that the test is passing in this case!

File: certbot-nginx/tests/obj_test.py [hunk 2]
```diff
@@ -168,7 +168,7 @@ def test_eq(self):
 
         self.assertEqual(vhost1b, self.vhost1)
         self.assertEqual(str(vhost1b), str(self.vhost1))
-        self.assertFalse(vhost1b == 1234)
+        self.assertEqual(vhost1b, 1234)
```
Comment (adferrand) [inline certbot-nginx/tests/obj_test.py:171]: This should be `assertNotEqual`, what is going on ...

**Review by adferrand — CHANGES_REQUESTED** [review]
Almost LGTM! But your modifications showed that two tests are passing while the assertion logic is inverted. I do not understand what is going on yet.

**Comment by adferrand** [comment]
Ah, I understand now. We are merging on to the `ecdsa` branch, so tests are not run. I will do it manually on your PR once you have fixed the two assertions.

File: certbot-nginx/tests/obj_test.py (hunk 2 above)
Comment (atombrella) [reply certbot-nginx/tests/obj_test.py]: Hmm. I've updated it. Hopefully this emphasizes my point that those asserts can be more difficult to read ...

Comment (adferrand) [reply certbot-nginx/tests/obj_test.py]: To be clear, the "what is going on" was about the fact that the tests were "passing", until I understood that they were not run. I completely agree with you that several assertions were double negation, so quite difficult to read properly.

**Comment by adferrand** [comment]
@atombrella I run the PR test suite as a final check.

**Review by adferrand — APPROVED** [review]
LGTM!

**Pull request merged**